common/models/data-source-definition.js

```diff
@@ -293,11 +293,16 @@ function ready(DataSourceDefinition) {
     http: { verb: 'POST' },
   });
 
+  // XXX(sam) This could really use a comment... it looks like the purpose here
+  // is not co-processing, but to workaround the fact that so much of what
+  // loopback does depends on the CWD from which its loaded?
   DataSourceDefinition.prototype.invokeMethodInWorkspace = function(methodName) {
     // TODO(bajtos) We should ensure there is never more than one instance
     // of this code running at any given time.
     var isDone = false;
     var self = this;
+    // XXX(sam) why not call slice as (arguments, 1) instead of shifting off
+    // arguments[0] later?
```

Comment:
I personally don't mind too much.

Note that we are doing more than just `args.shift()`, there is also `args.pop()`. I think the idea here is that we start with `args` as a write-able Array copy of our arguments that we modify later as needed.